presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -81,16 +81,27 @@ public ScheduleResult schedule()
         }
 
         CompletableFuture<?> blocked = CompletableFuture.completedFuture(null);
+        ScheduleResult.BlockedReason blockedReason = null;
+        int splitsScheduled = 0;
         while (!sourcePartitionedSchedulers.isEmpty()) {
             ScheduleResult schedule = sourcePartitionedSchedulers.peek().schedule();
+            splitsScheduled += schedule.getSplitsScheduled();
             blocked = schedule.getBlocked();
+            if (schedule.getBlockedReason().isPresent()) {
+                blockedReason = schedule.getBlockedReason().get();
```

Comment:
I think you need to overwrite `blockedReason` every loop or you will transfer the blocked reason from one source to the next